x-pack/plugins/security_solution/public/resolver/test_utilities/simulator/mock_resolver.tsx

```diff
@@ -0,0 +1,109 @@
+import { SideEffectContext } from '../../view/side_effect_context';
+import { sideEffectSimulatorFactory } from '../../view/side_effect_simulator_factory';
+
+type MockResolverProps = {
+  /**
+   * Used to simulate a raster width. Defaults to 1600.
+   */
+  rasterWidth?: number;
+  /**
+   * Used to simulate a raster height. Defaults to 1200.
+   */
+  rasterHeight?: number;
+  /**
+   * Used for the `KibanaContextProvider`
+   */
+  coreStart: CoreStart;
+  /**
+   * Used for `react-router`.
+   */
+  history: React.ComponentProps<typeof Router>['history'];
+  /** Pass a resolver store. See `storeFactory` and `mockDataAccessLayer` */
+  store: Store<ResolverState, ResolverAction>;
+  /**
+   * All the props from `ResolverWithoutStore` can be passed. These aren't defaulted to anything (you might want to test what happens when they aren't present.)
+   */
+} & ResolverProps;
+
+/**
+ * This is a mock Resolver component. It is intended to be used with `enzyme` tests via the `Simulator` class. It wraps Resolver in the required providers:
+ *  * `i18n`
+ *  * `Router` using a provided `History`
+ *  * `SideEffectContext.Provider` using side effect simulator it creates
+ *  * `KibanaContextProvider` using a provided `CoreStart` instance
+ *  * `react-redux`'s `Provider` using a provided `Store`.
+ *
+ * Resolver needs to measure its size in the DOM. The `SideEffectSimulator` instance can fake the size of an element.
+ * However in tests, React doesn't have good DOM reconciliation and the root element is often swapped out. When the
+ * element is replaced, the fake dimensions stop being applied. In order to get around this issue, this component will
+ * trigger a simulated resize on the root node reference any time it changes. This simulates the layout process a real
+ * browser would do when an element is attached to the DOM.
+ */
+export const MockResolver = React.memo((props: MockResolverProps) => {
+  const [resolverElement, setResolverElement] = useState<HTMLDivElement | null>(null);
+
+  // Get a ref to the underlying Resolver element so we can resize.
+  // Use a callback function because the underlying DOM node can change. In fact, `enzyme` seems to change it a lot.
+  const resolverRef = useCallback((element: HTMLDivElement | null) => {
+    setResolverElement(element);
+  }, []);
+
+  const simulator: SideEffectSimulator = useMemo(() => sideEffectSimulatorFactory(), []);
+
+  // Resize the Resolver element to match the passed in props. Resolver is size dependent.
+  useEffect(() => {
+    if (resolverElement) {
+      const size: DOMRect = {
+        width: props.rasterWidth ?? 1600,
+        height: props.rasterHeight ?? 1200,
+        x: 0,
+        y: 0,
+        bottom: 0,
+        left: 0,
+        top: 0,
+        right: 0,
+        toJSON() {
+          return this;
+        },
+      };
+      simulator.controls.simulateElementResize(resolverElement, size);
```

Comment:
TODO, idea for next time: `simulateElementResize` takes a query instead of an element reference. when responding to `getBoundingBox whatever` check if the element matches the query and if so return the fake size.